File: SManagement.py
class Teacher:
    def __init__(self, name, username, password):
        self.name = name
        self.username = username
        self.password = password
        self.is_logged_in = False

    def get_name(self):
        return self.name

    def get_username(self):
        return self.username

    def get_password(self):
        return self.password

    def set_username(self, username):
        self.username = username
        print("Username updated successfully")

    def set_password(self, password):
        self.password = password
        print("Password updated successfully")

    def set_logged_in(self, is_logged_in):
        self.is_logged_in = is_logged_in

    def is_logged_in_func(self):
        return self.is_logged_in

    def login(self, username, password):
        return self.username == username and self.password == password

    def logout(self):
        self.is_logged_in = False
        print("Logged out successfully")


class Student:
    def __init__(self, name, reg_no, age, marks):
        self.name = name
        self.reg_no = reg_no
        self.age = age
        self.marks = marks

    def get_name(self):
        return self.name

    def get_reg_no(self):
        return self.reg_no

    def get_age(self):
        return self.age

    def get_marks(self):
        return self.marks

    def set_marks(self, marks):
        self.marks = marks
        print("Marks updated successfully")

    def display_details(self):
        print(f"Name: {self.name}, Reg No: {self.reg_no}, Age: {self.age}, Marks: {self.marks}")


class SManagement:
    students = []
    teachers = []
    current_teacher = None

    @staticmethod
    def register():
        name = input("Enter the name of the teacher - ")
        username = input("Enter the username - ")
        password = input("Enter the password - ")
        SManagement.teachers.append(Teacher(name, username, password))
        print("Teacher registered successfully")

    @staticmethod
    def login():
        username = input("Enter the username - ")
        password = input("Enter password - ")
        for teacher in SManagement.teachers:
            if teacher.login(username, password):
                print("Login successful")
                teacher.set_logged_in(True)
                SManagement.current_teacher = teacher
                return
        print("Invalid username or password")

    @staticmethod
    def add_student():
        if SManagement.current_teacher is None or not SManagement.current_teacher.is_logged_in_func():
            print("Please login first")
            return

        name = input("Enter the name of the student - ")
        reg_no = int(input("Enter the registration number - "))
        age = int(input("Enter the age - "))
        marks = float(input("Enter the marks - "))
        SManagement.students.append(Student(name, reg_no, age, marks))
        print("Student added successfully")

    @staticmethod
    def view_student():
        if SManagement.current_teacher is None or not SManagement.current_teacher.is_logged_in_func():
            print("Please login first")
            return

        reg_no = int(input("Enter the registration number - "))
        found = False
        for student in SManagement.students:
            if student.get_reg_no() == reg_no:
                student.display_details()
                found = True
                break
        if not found:
            print("Student not found")

    @staticmethod
    def update_marks():
        if SManagement.current_teacher is None or not SManagement.current_teacher.is_logged_in_func():
            print("Please login first")
            return

        reg_no = int(input("Enter the registration number - "))
        found = False
        for student in SManagement.students:
            if student.get_reg_no() == reg_no:
                marks = float(input("Enter new marks - "))
                student.set_marks(marks)
                found = True
                break
        if not found:
            print("Student not found")

    @staticmethod
    def delete_student():
        if SManagement.current_teacher is None or not SManagement.current_teacher.is_logged_in_func():
            print("Please login first")
            return

        reg_no = int(input("Enter the registration number - "))
        found = False
        for student in SManagement.students:
            if student.get_reg_no() == reg_no:
                SManagement.students.remove(student)
                print("Student deleted successfully")
                found = True
                break
        if not found:
            print("Student not found")

    @staticmethod
    def logout():
        if SManagement.current_teacher is not None:
            SManagement.current_teacher.logout()
            SManagement.current_teacher = None
        else:
            print("No teacher is currently logged in")
    @staticmethod
    def main():
        while True:
            print("\n=== Student Management System ===")
            print("1. Login as Teacher")
            print("2. Register New Teacher")
            print("3. Add Student")
            print("4. View Student")
            print("5. Update Marks")
            print("6. Delete Student")
            print("7. Logout")
            print("8. Exit")
            choice = int(input("Enter your choice: "))
            if choice == 1:
                SManagement.login()
            elif choice == 2:
                SManagement.register()
            elif choice == 3:
                SManagement.add_student()
            elif choice == 4:
                SManagement.view_student()
            elif choice == 5:
                SManagement.update_marks()
            elif choice == 6:
                SManagement.delete_student()
            elif choice == 7:
                SManagement.logout()
            elif choice == 8:
                print("Exiting the program")
                break
            else:
                print("Invalid choice")
if __name__ == "__main__":
    SManagement.main()

File: SManagement.java
import java.util.*;

class Teacher {
    private String name;
    private String username;
    private String password;
    private boolean isLoggedIn;

    public Teacher(String name, String username, String password) {
        this.name = name;
        this.username = username;
        this.password = password;
        this.isLoggedIn = false;
    }

    public String getName() {
        return name;
    }

    public String getUsername() {
        return username;
    }

    public String getPassword() {
        return password;
    }

    public void setUsername(String username) {
        this.username = username;
        System.out.println("Username updated successfully");
    }

    public void setPassword(String password) {
        this.password = password;
        System.out.println("Password updated successfully");
    }

    public void setLoggedIn(boolean isLoggedIn) {
        this.isLoggedIn = isLoggedIn;
    }

    public boolean isLoggedIn() {
        return isLoggedIn;
    }

    public boolean login(String username, String password) {
        return this.username.equals(username) && this.password.equals(password);
    }

    public void logout() {
        this.isLoggedIn = false;
        System.out.println("Logged out successfully");
    }
}

class Student {
    private String name;
    private int regNo;
    private int age;
    private double marks;

    public Student(String name, int regNo, int age, double marks) {
        this.name = name;
        this.regNo = regNo;
        this.age = age;
        this.marks = marks;
    }

    public String getName() {
        return name;
    }

    public int getRegNo() {
        return regNo;
    }

    public int getAge() {
        return age;
    }

    public double getMarks() {
        return marks;
    }

    public void setMarks(double marks) {
        this.marks = marks;
        System.out.println("Marks updated successfully");
    }

    public void displayDetails() {
        System.out.println("Name: " + name + ", Reg No: " + regNo + ", Age: " + age + ", Marks: " + marks);
    }
}

public class SManagement {
    private static final Scanner sc = new Scanner(System.in);
    private static List<Student> students = new ArrayList<>();
    private static List<Teacher> teachers = new ArrayList<>();
    private static Teacher currentTeacher = null;

    public static void register() {
        System.out.println("Enter the name of the teacher - ");
        String name = sc.nextLine();
        System.out.println("Enter the username - ");
        String username = sc.nextLine();
        System.out.println("Enter the password - ");
        String password = sc.nextLine();
        teachers.add(new Teacher(name, username, password));
        System.out.println("Teacher registered successfully");
    }

    public static void login() {
        System.out.println("Enter the username - ");
        String username = sc.nextLine();
        System.out.println("Enter password - ");
        String password = sc.nextLine();
        for (Teacher teacher : teachers) {
            if (teacher.login(username, password)) {
                System.out.println("Login successful");
                teacher.setLoggedIn(true);
                currentTeacher = teacher;
                return;
            }
        }
        System.out.println("Invalid username or password");
    }

    public static void addStudent() {
        if (currentTeacher == null || !currentTeacher.isLoggedIn()) {
            System.out.println("Please login first");
            return;
        }
        
        System.out.println("Enter the name of the student - ");
        String name = sc.nextLine();
        System.out.println("Enter the registration number - ");
        int regNo = sc.nextInt();
        System.out.println("Enter the age - ");
        int age = sc.nextInt();
        System.out.println("Enter the marks - ");
        double marks = sc.nextDouble();
        students.add(new Student(name, regNo, age, marks));
        System.out.println("Student added successfully");
    }

    public static void viewStudent() {
        if (currentTeacher == null || !currentTeacher.isLoggedIn()) {
            System.out.println("Please login first");
            return;
        }

        System.out.println("Enter the registration number - ");
        int regNo = sc.nextInt();

        boolean found = false;
        for (Student student : students) {
            if (student.getRegNo() == regNo) {
                student.displayDetails();
                found = true;
                break;
            }
        }

        if (!found) {
            System.out.println("Student not found");
        }
    }

    public static void updateMarks() {
        if (currentTeacher == null || !currentTeacher.isLoggedIn()) {
            System.out.println("Please login first");
            return;
        }
        System.out.println("Enter the registration number - ");
        int regNo = sc.nextInt();
        boolean found = false;
        for (Student student : students) {
            if (student.getRegNo() == regNo) {
                System.out.println("Enter new marks - ");
                double marks = sc.nextDouble();
                student.setMarks(marks);
                found = true;
                break;
            }
        }
        if (!found) {
            System.out.println("Student not found");
        }
    }

    public static void deleteStudent() {
        if (currentTeacher == null || !currentTeacher.isLoggedIn()) {
            System.out.println("Please login first");
            return;
        }
        System.out.println("Enter the registration number - ");
        int regNo = sc.nextInt();
        boolean found = false;
        Iterator<Student> iterator = students.iterator();
        while (iterator.hasNext()) {
            Student student = iterator.next();
            if (student.getRegNo() == regNo) {
                iterator.remove();
                System.out.println("Student deleted successfully");
                found = true;
                break;
            }
        }
        if (!found) {
            System.out.println("Student not found");
        }
    }

    public static void logout() {
        if (currentTeacher != null) {
            currentTeacher.logout();
            currentTeacher = null;
        } else {
            System.out.println("No teacher is currently logged in");
        }
    }

    public static void main(String[] args) {
        while (true) {
            System.out.println("\n=== Student Management System ===");
            System.out.println("1. Login as Teacher");
            System.out.println("2. Register New Teacher");
            System.out.println("3. Add Student");
            System.out.println("4. View Student");
            System.out.println("5. Update Marks");
            System.out.println("6. Delete Student");
            System.out.println("7. Logout");
            System.out.println("8. Exit");
            System.out.print("Enter your choice: ");
            int choice = sc.nextInt();
            sc.nextLine(); // Clear buffer
            
            switch (choice) {
                case 1:
                    login();
                    break;
                case 2:
                    register();
                    break;
                case 3:
                    addStudent();
                    break;
                case 4:
                    viewStudent();
                    break;
                case 5:
                    updateMarks();
                    break;
                case 6:
                    deleteStudent();
                    break;
                case 7:
                    logout();
                    break;
                case 8:
                    System.out.println("Exiting the program");
                    sc.close();
                    System.exit(0);
                    break;
                default:
                    System.out.println("Invalid choice");
                    break;
            }
        }
    }
}
